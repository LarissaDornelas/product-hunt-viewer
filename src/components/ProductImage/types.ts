export enum ImageSize {
  SMALL = "15px",
  MEDIUM = "40px",
  LARGE = "80px",
}

export interface IProductImage {
  imageUrl: string;
  size: ImageSize;
}
