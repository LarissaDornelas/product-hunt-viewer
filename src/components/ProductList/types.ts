export interface IProductItem {
  id: string;
  name: string;
  description: string;
  votesCount: string;
  thumbnail: string;
}

export interface IProductList {
  data: IProductItem[];
  loading: boolean;
  handleFetchMore: () => void;
}
