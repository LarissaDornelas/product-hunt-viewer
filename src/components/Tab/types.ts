import { Order } from "hooks/posts/types";

export interface ITabItem {
  title: string;
  active: boolean;
  value: Order;
}
export interface ITab {
  tabItems: ITabItem[];
  toggleTab: (index: number) => void;
}
