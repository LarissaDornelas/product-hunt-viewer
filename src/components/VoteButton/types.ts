export interface IVoteButton {
  votes: number;
}
