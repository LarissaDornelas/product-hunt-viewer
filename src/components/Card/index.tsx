import React from "react";

import { Container } from "./styles";

const Card: React.FC = (props) => {
  const { children } = props;
  return <Container data-testid="card-container">{children}</Container>;
};

export { Card };
