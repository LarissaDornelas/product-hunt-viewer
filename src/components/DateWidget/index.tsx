import React from "react";
import { format } from "date-fns";

import { Container } from "./styles";

const DateWidget: React.FC = () => {
  return (
    <Container data-testid="date-widget-container">
      <p>{format(new Date(), "'Today', Lo LLL ")}</p>
    </Container>
  );
};

export { DateWidget };
