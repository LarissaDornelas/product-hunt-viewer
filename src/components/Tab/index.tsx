import React from "react";

import { ITab, ITabItem } from "./types";
import { Container, TabItem } from "./styles";

const Tab: React.FC<ITab> = (props) => {
  const { tabItems, toggleTab } = props;

  return (
    <Container data-testid="tab-container">
      {tabItems.map((item: ITabItem, index: number) => (
        <TabItem
          data-testid="tab-item"
          key={index}
          active={item.active}
          onClick={() => toggleTab(index)}
        >
          {item.title}
        </TabItem>
      ))}
    </Container>
  );
};

export { Tab };
