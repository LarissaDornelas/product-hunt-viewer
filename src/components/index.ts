import { ProfileImage } from "./ProfileImage";
import { DateWidget } from "./DateWidget";
import { SearchButton } from "./SearchButton";
import { Tab } from "./Tab";
import { Card } from "./Card";

export { ProfileImage, DateWidget, SearchButton, Tab, Card };
