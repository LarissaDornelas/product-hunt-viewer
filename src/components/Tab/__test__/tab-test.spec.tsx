import React from "react";
import ReactDOM from "react-dom";
import {
  render,
  cleanup,
  getByTestId,
  getAllByTestId,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import { Tab } from "components";
import { Order } from "hooks/posts/types";

describe("Component: Tab", () => {
  afterEach(cleanup);

  const tabMock = {
    tabItems: [
      { title: "popular", active: true, value: Order.RANKING },
      { title: "newest", active: false, value: Order.NEWEST },
    ],
    toggleTab: jest.fn(),
  };

  it("Should render without crashing", () => {
    const container = document.createElement("div");
    ReactDOM.render(<Tab {...tabMock}></Tab>, container);
  });

  it("Should render Tab correctly", () => {
    const { getByTestId } = render(<Tab {...tabMock}></Tab>);

    expect(getByTestId("tab-container")).toHaveTextContent("popularnewest");
  });

  it("Should click tab", () => {
    const { getAllByTestId } = render(<Tab {...tabMock}></Tab>);
    const tabItem = getAllByTestId("tab-item");

    tabItem[0].click();

    expect(tabMock.toggleTab).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Tab {...tabMock}></Tab>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
