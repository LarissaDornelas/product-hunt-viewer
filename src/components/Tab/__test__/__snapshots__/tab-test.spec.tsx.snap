// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tab matches snapshot 1`] = `
<div
  className="sc-hKgILt lfwwFe"
  data-testid="tab-container"
>
  <div
    className="sc-eCssSg jksizx"
    data-testid="tab-item"
    onClick={[Function]}
  >
    popular
  </div>
  <div
    className="sc-eCssSg hbFUcZ"
    data-testid="tab-item"
    onClick={[Function]}
  >
    newest
  </div>
</div>
`;
